src/app/admin/collectionCentresTable/CollectionCentresTable.tsx

```diff
@@ -99,6 +173,111 @@ const CollectionCentresTable: React.FC = () => {
         };
     }, [getCollectionCentresForTable]);
 
+    const handleModalSaveClick = async (): Promise<void> => {
+        if (timeSlotModalData === null) {
+            return;
+        }
+        const { error: updateTimeSlotError } =
+            await updateDbCollectionCentreTimeSlots(timeSlotModalData);
+        const baseAuditLog = getBaseAuditLogForCollectionCentreTimeSlots(
+            "update collection centre time slots",
+            timeSlotModalData
+        );
+
+        if (updateTimeSlotError) {
+            setTimeSlotModalErrorMessage(
+                `Failed to update the collection centre time slots. Log ID: ${updateTimeSlotError.logId}`
+            );
+            await sendAuditLog({
+                ...baseAuditLog,
+                wasSuccess: false,
+                logId: updateTimeSlotError.logId,
+            });
+        }
+
+        await sendAuditLog({ ...baseAuditLog, wasSuccess: true });
+        setTimeSlotModalIsOpen(false);
+    };
+
+    const handleAddSlotClick = async (): Promise<void> => {
+        setEditableIsShown(true);
+    };
+
+    const checkIfSlotExists = (
+        existingTimeSlotData: FormattedTimeSlotsWithPrimaryKey,
+        newTimeSlot: FormattedTimeSlot
+    ): boolean => {
+        return existingTimeSlotData.timeSlots.some((slot) => slot.time === newTimeSlot.time);
+    };
+
+    const addNewTimeSlotToTimeSlotModalData = (
+        existingTimeSlotData: FormattedTimeSlotsWithPrimaryKey,
+        newTimeSlot: FormattedTimeSlot
+    ): void => {
+        const newTimeSlotArray = [...existingTimeSlotData.timeSlots, newTimeSlot];
+        newTimeSlotArray.sort((slot1, slot2) => slot1.time.localeCompare(slot2.time));
+
+        const updatedTimeSlotModalData: FormattedTimeSlotsWithPrimaryKey = {
+            ...existingTimeSlotData,
+            timeSlots: newTimeSlotArray,
+        };
+
+        setTimeSlotModalData(updatedTimeSlotModalData);
+    };
+
+    const handleSaveSlotClick = async (): Promise<void> => {
+        setTimeSlotModalErrorMessage(null);
+        if (timeSlotModalData === null || collectionTimeSlotValue === undefined) {
```

Comment:
I think it may be initially not setting the value? i.e. if I press add, it by default fills it up with the current time, however nothing happens if I press save. If I change the value myself to the time (i.e. interact with the field and update values) then it works